pkg/dashboard/store/metrics/metrics.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2021 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package metrics
+
+import (
+	"context"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"go.uber.org/fx"
+
+	"github.com/chaos-mesh/chaos-mesh/pkg/dashboard/core"
+)
+
+const chaosDashboardSubsystem = "chaos_dashboard"
+
+// Collector implements prometheus.Collector interface
+type Collector struct {
+	experimentStore core.ExperimentStore
+	scheduleStore   core.ScheduleStore
+	workflowStore   core.WorkflowStore
+
+	archivedExperiments *prometheus.GaugeVec
+	archivedSchedules   *prometheus.GaugeVec
+	archivedWorkflows   *prometheus.GaugeVec
+}
+
+// NewCollector initializes metrics and collector
+func NewCollector(experimentStore core.ExperimentStore, scheduleStore core.ScheduleStore, workflowStore core.WorkflowStore) *Collector {
+	return &Collector{
+		experimentStore: experimentStore,
+		scheduleStore:   scheduleStore,
+		workflowStore:   workflowStore,
+		archivedExperiments: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Subsystem: chaosDashboardSubsystem,
+			Name:      "archived_experiments",
+			Help:      "Total number of archived chaos experiments",
+		}, []string{"namespace", "type"}),
+		archivedSchedules: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Subsystem: chaosDashboardSubsystem,
+			Name:      "archived_schedules",
+			Help:      "Total number of archived chaos schedules",
+
+			// TODO: labels
```

Comment:
I think there are no other labels that could be attached to this metric. Maybe we could remove this line?

Reply:
Oh, I forgot to delete this line.

Reply:
Done.